Userland/Libraries/LibJS/Runtime/Temporal/PlainDatePrototype.cpp

```diff
@@ -552,20 +552,18 @@ JS_DEFINE_NATIVE_FUNCTION(PlainDatePrototype::since)
     auto* until_options = TRY(merge_largest_unit_option(global_object, *options, move(*largest_unit)));
 
     // 15. Let result be ? CalendarDateUntil(temporalDate.[[Calendar]], temporalDate, other, untilOptions).
-    auto* result = TRY(calendar_date_until(global_object, temporal_date->calendar(), temporal_date, other, *until_options));
+    auto* duration = TRY(calendar_date_until(global_object, temporal_date->calendar(), temporal_date, other, *until_options));
 
-    // 16. If smallestUnit is "day" and roundingIncrement = 1, then
-    if (*smallest_unit == "day"sv && rounding_increment == 1) {
-        // a. Return ? CreateTemporalDuration(−result.[[Years]], −result.[[Months]], −result.[[Weeks]], −result.[[Days]], 0, 0, 0, 0, 0, 0).
-        return TRY(create_temporal_duration(global_object, -result->years(), -result->months(), -result->weeks(), -result->days(), 0, 0, 0, 0, 0, 0));
-    }
+    DurationRecord result { duration->years(), duration->months(), duration->weeks(), duration->days(), 0, 0, 0, 0, 0, 0 };
```

Comment:
What's the point of this? :thinking:

Reply:
`result` from step 15 is a `Duration*`, `result` from the optional step 16a a `DurationRecord` - both have the internal slot accessed unconditionally in step 17, but C++ can't do that, so we copy them to a struct of the same type.

Reply:
Ah, unfortunate, let's hope the compiler is smarter than me :)